packages/thumbprint-react/components/Calendar/index.tsx

```diff
@@ -170,39 +145,41 @@ export default function Calendar({
 
     const selectedDays = normaliseValue(value);
 
-    const modifiers: Partial<Modifiers> = {};
+    const modifiers: Record<string, Matcher> = {};
+    const modifiersClassNames: Record<Modifier, string> = {};
 
     if (typeof daysThemeDotIndicator === 'function') {
         modifiers['theme-dot'] = daysThemeDotIndicator;
+        modifiersClassNames['theme-dot'] = 'rdp-day--theme-dot';
     }
 
     if (typeof daysThemeStrikeout === 'function') {
         modifiers['theme-strikeout'] = daysThemeStrikeout;
+        modifiersClassNames['theme-strikeout'] = 'rdp-day--theme-strikeout';
     }
 
     return (
         <div className={styles.root}>
             <DayPicker
-                disabledDays={disabledDays || undefined}
+                disabled={disabledDays || undefined}
                 fromMonth={get(disabledDays, 'before', null)}
                 toMonth={lastMonth}
-                month={month || selectedDays[0]}
-                initialMonth={month || selectedDays[0]}
-                selectedDays={selectedDays}
+                defaultMonth={
+                    month ||
+                    (selectedDays && selectedDays.length > 0 ? selectedDays[0] : new Date())
+                }
+                selected={selectedDays}
                 onMonthChange={onMonthChange}
                 modifiers={modifiers}
-                onDayClick={(day, { selected, disabled }): void => {
-                    if (disabled) {
-                        return;
-                    }
-
+                modifiersClassNames={modifiersClassNames}
+                onDayClick={(day): void => {
                     let newSelectedDays = normaliseValue(value);
 
                     if (allowMultiSelection) {
-                        if (selected) {
-                            const selectedIndex = findIndex(newSelectedDays, selectedDay =>
-                                DateUtils.isSameDay(selectedDay, day),
-                            );
+                        const selectedIndex = findIndex(newSelectedDays, selectedDay =>
```

Comment:
Note: Only receiving `day` input from `onDayClick ` , so checking is selected day click or not manually.

Reply:
We used to return early if the day was `disabled`. Is that an issue?

Reply:
In v7, for disabled date click, we were onDayClick() was being called with `disabled: true`. So we were manually return from this function for that case.  But,  In v8, for disabled date click, onDayClick() is not being called, So, no need to  return from it for `disabled` state.

And good point. I checked prop types again. We are receiving `selected` state in second param of onDayClick(), so removing manual selected day check and using `selected` param. (Apologies, I missed it initially.)